keps/sig-node/4639-oci-volume-source/README.md

```diff
@@ -0,0 +1,883 @@
+so that I can streamline my CI/CD pipeline. I want to package this file in an OCI object to take advantage of OCI distribution.
+
+#### Story 3
+
+As a data scientist, MLOps engineer, or AI developer, I want to mount large language models or machine learning models in a pod alongside a model-server, so that I can efficiently serve the models
+without including them in the container image. I want to package these models in an OCI object to take advantage of OCI distribution and ensure
+efficient model deployment.
+
+### Notes/Constraints/Caveats (Optional)
+
+- This enhancement assumes that the cluster has access to the OCI registry.
+- The implementation must handle image pull secrets and other registry authentication mechanisms.
+- Performance considerations must be taken into account, especially for large images or artifacts.
+
+### Vocabulary: OCI Images, Artifacts, and Objects
+
+**OCI Image:**
+   - A container image that conforms to the Open Container Initiative (OCI) Image Specification.
+     It includes a filesystem bundle and metadata required to run a container.
+   - Consists of multiple layers (each layer being a tarball), a manifest (which lists the layers), and a config file
+     (which provides configuration data such as environment variables, entry points, etc.).
+   - **Use Case:** Used primarily for packaging and distributing containerized applications.
+
+**OCI Artifact:**
+   - An artifact describes any content that is stored and distributed using the OCI image format.
+     It includes not just container images but also other types of content like Helm charts, WASM modules, machine learning models, etc.
+   - Artifacts use the same image manifest and layer structure but may contain different types of data
+     within those layers. The artifact manifest can have media types that differ from those in standard container images.
+   - **Use Case:** Allows the distribution of non-container content using the same infrastructure and tools developed for OCI images.
+
+**OCI Object:**
+   - Umbrella term encompassing both OCI images and OCI artifacts. It represents any object that conforms to the OCI specifications for storage and distribution.
+
+### Risks and Mitigations
+
+- **Security Risks:** Allowing direct mounting of OCI images introduces potential attack vectors. Mitigation includes thorough security reviews and
+  limiting access to trusted registries. Limiting to OCI artifacts (non-runnable content) or read-only mode may lessen the security risk.
+- **Compatibility Risks:** Ensure compatibility with existing features.
+- **Performance Risks:** Large images or artifacts could impact performance. Mitigation includes optimizations in the implementation and providing
+  guidance on best practices for users.
+
+## Design Details
+
+The new `VolumeSource` will be defined in the Kubernetes API, and the implementation will involve updating components (CRI, Kubelet, Scheduler)
```

Comment:
Is the intent that the container runtime would be responsible for pulling the image/artifact and materializing it (e.g., handling the layer union)?  It seems like that model would allow reuse of existing logic around pull, credentials, layer backends, etc and would allow reuse of any signing integration.

Reply:
Yes, that's the intent based on conversations so far. That detail should be captured here.

Reply:
More thoughts on things that are involved in the existing image management area that are relevant here:
* pull
* credential management (image pull secrets and credential providers)
* garbage collection
* pod annotations/runtime handler passthrough
* lazy-loading implementations

Reply:
Yes, we definitely want to retain all those features that we already use for images :+1:

Reply:
I added more detail in the section. PTAL again and open new discussion points if required.